### ISSUE TITLE
[css-grid] Compiler breaks after switching Areas in Grid Layouts

### ISSUE THREAD

**Issue by gisu** [postcss/autoprefixer#1096]
Testing the new autoprefixer right now. We're in trouble. In the grid layout, I swap elements about the breakpoints (all this also works only with IE the wrong sequence of breakpoints created).

```scss
&__inner {
  display: grid;
  grid-gap: 0;
  grid-template:
    'oct log qu1 qu1 qu1' 65px
    'slo slo slo slo slo' auto /
      65px 65px 1fr 1fr 1fr;

  @include break('m') {
    grid-gap: 10px 20px;
    grid-template:
      'log met qu2' 20px
      'log lin lin' auto
      'log nav nav' auto /
        200px 1fr 110px;
  }

  @include break('l') {
    grid-gap: 10px 50px;
    grid-template:
      'log met soc' 20px
      'log lin lin' auto
      'log nav qu2' auto /
      290px 1fr 110px;
  }
}

&__log { grid-area: log; }

&__met { grid-area: met; }

.c-pageSlogan { grid-area: slo; }

&__lin { grid-area: lin; }

.c-offCanvasTrigger {
  grid-area: oct;
  @include break('m') {
    display: none;
  }
}

.c-quicklinksMobile {
  grid-area: qu1;
  @include break('m') {
    display: none;
  }
}

.c-quicklinksDesktop {
  display: none;
  grid-area: qu2;
  @include break('m') {
    display: flex;
  }
}

.c-mainnav {
  display: none;
  grid-area: nav;
  @include break('m') {
    display: block;
  }
}

.c-socialNetworks {
  display: none;
  grid-area: soc;
  @include break('l') {
    display: flex;
  }
}
```

Then creates an error in the console

```bash
events.js:165
      throw er; // Unhandled 'error' event
      ^
TypeError: Cannot read property 'row' of undefined
    at /Users/saschafuchs/Develop/private/gastrocard/site/node_modules/autoprefixer/lib/hacks/grid-utils.js:428:42
    at /Users/saschafuchs/Develop/private/gastrocard/site/node_modules/postcss-scss/node_modules/postcss/lib/container.es6:163:28
    at /Users/saschafuchs/Develop/private/gastrocard/site/node_modules/postcss-scss/node_modules/postcss/lib/container.es6:109:26
    at Rule.each (/Users/saschafuchs/Develop/private/gastrocard/site/node_modules/postcss-scss/node_modules/postcss/lib/container.es6:77:22)
    at Rule.walk (/Users/saschafuchs/Develop/private/gastrocard/site/node_modules/postcss-scss/node_modules/postcss/lib/container.es6:108:21)
    at Rule.walkDecls (/Users/saschafuchs/Develop/private/gastrocard/site/node_modules/postcss-scss/node_modules/postcss/lib/container.es6:161:25)
...
Emitted 'error' event at:
    at Transform.onerror (/Users/saschafuchs/Develop/private/gastrocard/site/node_modules/readable-stream/lib/_stream_readable.js:640:52)
    at Transform.emit (events.js:180:13)
    at onwriteError (_stream_writable.js:431:12)
    at onwrite (_stream_writable.js:453:5)
    at Transform.afterTransform (_stream_transform.js:89:3)
    at Immediate.<anonymous> (/Users/saschafuchs/Develop/private/gastrocard/site/node_modules/gulp-postcss/index.js:93:9)
    at runCallback (timers.js:763:18)
    at tryOnImmediate (timers.js:734:5)
    at processImmediate (timers.js:716:5)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gastrotopcard@0.0.51 dev: `cross-env NODE_ENV=development gulp`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the gastrotopcard@0.0.51 dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

The error occurs when I swap areas via the breakpoints. If I just change the position, there's no mistake. 

But it's only now with the new autoprefixer, last week it was still running.

**Comment by gisu** [comment]
I marked the affected line for you. Grep has shown me several autoprefixers (is also not quite small the project in which I try it), but they are not affected by the error.

```js
node.walkDecls('grid-template', function (d) {
      var template = parseTemplate({ decl: d, gap: getGridGap(d) });
      var comparedArea = template.areas[areaName];
      if (area.row.span !== comparedArea.row.span) {      // <<<<< Line 428
        overrideValues[0] = true;
      } else if (area.column.span !== comparedArea.column.span) {
        overrideValues[1] = true;
      }

      return undefined;
    });
```

**Comment by bogdan0083** [comment]
Looks like it is my fuck up, sorry. I'll try to fix the bug asap

**Comment by bogdan0083** [comment]
Fixed: #1097

**Comment by ai** [comment]
I will release a fic on next Can I Use db release

**Comment by Dan503** [comment]
Do you know when the next caniuse DB release is scheduled for?

It seems to be holding up quite a few issues now :(

**Comment by ai** [comment]
There is a nice way to force Can I Use release. Just fix anything in their repo. Grammar, add links, anything.

**Comment by ai** [comment]
Released in 9.1.1
